### ISSUE TITLE
on login see: googlevoice.util.LoginError

### ISSUE THREAD

**Issue by dylnclrk** [pettazz/pygooglevoice#1]
when I try and run the sms.py example, it fails with error `<googlevoice.util.LoginError>`.
could it be that google has changed the login url?

**Comment by vangheem** [comment]
Same issue for me.

**Comment by twbarber** [comment]
Is this still being actively developed? I can begin looking into this issue tonight.

**Comment by vangheem** [comment]
The way this package is implemented, I don't it'll ever work with application specific passwords as google does not allow you to use those passwords to login to the web google voice site.

**Comment by twbarber** [comment]
http://www.androidpolice.com/2013/11/04/merging-of-google-voice-and-hangouts-will-result-shutting-down-all-3rd-party-voice-apps-in-may-2014/

It appears that this may not work anymore at all, sadly. The Voice API looks like it has been shut down. I wonder if there are alternatives.

**Comment by vangheem** [comment]
I don't think this uses an API. I think this simulates logging in through the browser and sending a text through the web interface. It's a hack :)

**Comment by pettazz** [comment]
That's 100% accurate. This was a hack to basically simulate login and scrape the web UI specifically because there was no API. I found a long-dead version on googlecode in like 2009 and fixed the existing bugs at the time and put it up here. But since then I think google logins have become so much more complicated and changed that the login stuff is now seriously broken. But I don't think the voice UI has changed significantly, so if you're up to it, feel free to make some PRs. But for now, I'm not actively developing this.

**Comment by vangheem** [comment]
Oh, it still works. Just doesn't work with application specific passwords--that was my problem.

**Comment by vangheem** [comment]
So if you have 2-factor auth on the google account you want to interface with, you'll need to do the whole sms auth code thing. Which means, you can't use this library to automate any api level stuff with an application specific generated password.

**Comment by vangheem** [comment]
And the reason it doesn't work with application passwords is a google restraint, not a library one. You simply can't login through the web UI with application passwords.

**Comment by twbarber** [comment]
Whoa, neat! I was unaware. @vangheem, I see what you're saying now. Would it be possible to work with a separate account strictly for your purpose without Two-Factor?

**Comment by vangheem** [comment]
Yes.

**Comment by The-Penultimate-Defenestrator** [comment]
I get a LoginError today when I didn't yesterday. Has anything changed?

**Comment by twbarber** [comment]
I can check tonight. Are you using the most recent version from this repository?

**Comment by The-Penultimate-Defenestrator** [comment]
Downloaded it yesterday morning, it should be.

**Comment by The-Penultimate-Defenestrator** [comment]
Great. I'm travelling right now, so I'm on a lot of different wifi networks. Last night there was a brief period in which logins didn't work on my hotel wifi but did using my phone's service (through the Pythonista iOS app). Now it doesn't work either through my phone or on the airport's wifi, nor from a computer in my house that I access through SSH.

**Comment by The-Penultimate-Defenestrator** [comment]
Solved! The issue was in the fact that I was signing in while traveling. The fact that I'm in Houston instead of NY was flagged as "suspicious activity" and therefore as soon as I logged in graphically I was prompted to confirm by entering what city I normally logged in from. This was also likely the difference between hotel wifi and my phone.
